lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -53,49 +244,60 @@ define((require, exports, module) ->
       { type: "rparen", value: "}" }
     ]
 
+    # empty constructor
     constructor: (@editor) ->
 
-    equalTokens: (token1, token2) ->
-      if token1? and token2?
-        return token1.type == token2.type and token1.value == token2.value
-      else return if token1? or token2? then false else true
+    @equalTokens: (token1, token2) ->
+      return token1.type == token2.type and token1.value == token2.value
 
     getEquationStart: (tokenIterator) ->
+      # if tokenIterator initialy is on the empty line, it's current token is null
+      if not tokenIterator.getCurrentToken()?
```

Comment:
Nope. It is only the case, when token iterator is **initially** on an empty line. If we have an empty line ahead of us during the iteration, it is just skipped.